### DISCUSSION TITLE
Tag hyperlink failing when blog is placed in subfolder

### DISCUSSION THREAD

**Discussion by hysonlee** [datenstrom/community#162] [Report a bug] [answered]
Hi,
I met an interesting problem, first introduce my file structure of `content` folder:

```
/
├content
    ├─1-home
    ├─2-note
    │   ├─3-blog
    │
    ├─4-wiki
    ├─5-about
```

The `blog` folder placed in a sub folder under `content`, when click the blog tag ,page not list the related tag of article(s), just go to the parent directory. For my this case, the URL of blog tag(s) should be `website/note/blog/tag:xxx/`, but the showing URL is  `website/note/tag:xxx/` when hover, after click the page display the the content of `website/note/`. Wiki pages also the same problem.

All the verification under the fresh Yellow installation and the version up to date, plugins just added `blog`,`wiki` and `breadcrumbs`.

Where I made a mistake or a possible bug.

**Comment by markseuffert** [comment]
Hi Hyson, good to hear from you again. 

I looked into the source code, here's what I found out. The blog makes the assumption to be in one of the top folders. When it creates links for tags it uses `$yellow->page->getParentTop()->getLocation()` to figure out where the blog main page is. Which works when the blog is in a top folder, but is wrong when the blog is placed in a subfolder. Exactly the same thing happens in blog and wiki.

Not sure if this is a bug or a feature. Should we give up the assumption that the blog/wiki main page is in one of the top folders? If yes, then we can replace `getParentTop()` with `getParent()` in snippets. What do you think?

**Comment by hysonlee** [comment]
I think no matter where the blog/wiki be located in and all the work fine that would be better. Of course the assumption is good for starter.

Thanks for your explanation， Mark.

**Comment by markseuffert** [comment]
Good point, I updated blog and wiki as you suggested. Does this work for you?

**Comment by markseuffert** [comment, marked as answer]
Looks like you are busy, Hyson. I close this issue in good belief that it's been solved.

**Comment by hysonlee** [comment]
Sorry for delaying response,these days are really busy, I'll have a try later.

**Comment by hysonlee** [comment]
I have tried the updated `blog` and `wiki` plugins, they are work now, premise is to manual define the Meta settings as`Template:blogpages` or`Template: wikipages` at the default `page.txt`.

I remember you have explained it to me that the Yellow program will automatically detect the folder name of `2-blog`/`3-wiki` where they are as their template, if changed their folder name then should manual setting  templates. In this case, `3-blog` under a sub folder same effect as change folder name.

**Comment by markseuffert** [comment]
Thanks for testing, Hyson.
